kolibri/core/assets/src/disconnection.js

```diff
@@ -0,0 +1,62 @@
+import { createTranslator } from 'kolibri.utils.i18n';
+
+export const trs = createTranslator('disconnectionSnackbars', {
+  disconnected: 'Disconnected from server. Will try to reconnect in { remainingTime }',
+  tryNow: 'Try now',
+  tryingToReconnect: 'Trying to reconnect…',
+  successfullyReconnected: 'Successfully reconnected!',
+});
+
+export function createTryingToReconnectSnackbar(store) {
+  store.dispatch('CORE_CREATE_SNACKBAR', {
+    text: trs.$tr('tryingToReconnect'),
+    backdrop: true,
+  });
+}
+
+let dynamicReconnectTime = 0;
+let timer = null;
+
+export function createDisconnectedSnackbar(store, beatCallback) {
+  // clear timers
+  clearTimer();
+  // set proper time
+  setDynamicReconnectTime(store.state.core.connection.reconnectTime);
+  // create snackbar
+  store.dispatch('CORE_CREATE_SNACKBAR', {
+    text: generateDisconnectedSnackbarText(),
+    actionText: trs.$tr('tryNow'),
+    actionCallback: () => beatCallback(),
```

Comment:
This could be just `actionCallback: beatCallback`, right?

Reply:
Yeah, don't need the wrapper function.